### PR TITLE
roachtest: use cgroupv2 API in disk-stalled roachtests

### LINKED ISSUE #112111: roachtest: update disk-stalled tests for ubuntu 22.04
Ubuntu images for roachtests were upgraded to 22.04 in [#111517](https://github.com/cockroachdb/cockroach/pull/111517). `disk-stalled` tests error out on Ubuntu 22.04, so they are currently still running on Ubuntu 20.04 (Focal Fossa) through a cluster.spec override. https://github.com/cockroachdb/cockroach/blob/1f86b1e9c6ef4c5d8cb6656cc432a56a5282124a/pkg/cmd/roachtest/tests/disk_stall.go#L49

We need to fix this test and remove the above override so it can run on Ubuntu 22.04.

The `disk-stalled` tests use `blkio` from `cgroups v1` to throttle IO, e.g.
https://github.com/cockroachdb/cockroach/blob/1f86b1e9c6ef4c5d8cb6656cc432a56a5282124a/pkg/cmd/roachtest/tests/disk_stall.go#L384-L390

However, Ubuntu 22 has switched to using `cgroups v2`, causing `cgroups v1` and thus `blkio` to no longer work. The above line now throws the following error:
`/bin/bash: line 1: /sys/fs/cgroup/blkio/blkio.throttle.write_bps_device: No such file or directory`

It is possible to make a custom build with `cgroups v1` enabled. 

```
# Enable cgroup v1, Cgroup v2 is now enabled by default. If you want to switch to cgroup v1 instead, you need to set the following kernel parameter:
 
systemd.unified_cgroup_hierarchy=0
```
However, this probably isn't worth the effort. Instead, we should be using [io.max](https://docs.kernel.org/admin-guide/cgroup-v2.html#io-interface-files) from cgroups v2, which will also let us throttle IO.

Jira issue: CRDB-32239

### PR DESCRIPTION
**Pull request by jbowens**
In recent versions of Ubuntu (eg, 22.04), the cgroupv2 API is the default.
Switch to using the new cgroupv2 API for limiting bandwidth to simulate disk
stalls, allowing this roachtest to run on the new Ubuntu versions.

Resolves #112111.
Epic: none
Release note: None

### CODE REVIEW

File: pkg/cmd/roachtest/tests/disk_stall.go [hunk 1]
```diff
@@ -383,16 +383,46 @@ func (s *cgroupDiskStaller) device() (major, minor int) {
 	}
 }
 
+type throughput struct {
+	limited        bool
+	bytesPerSecond int
```
Comment (DarrylWong) [inline pkg/cmd/roachtest/tests/disk_stall.go:388]: Nit: Not sure if we have a style rule against it, but does this need to be an int? If it were just a string we could set it to bytes or "max" accordingly and not need this throughput struct or the int to string conversion logic.

**Review by DarrylWong — APPROVED** [review]
Looks good thanks for doing this!

**Review by jbowens — COMMENTED** [review]
TFTR!

Updated to address an issue where the `Unstall` call could fail because the process already exited (and so the relevant dirs within `/sys/fs/cgroup/` no longer exist).

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/112628#-:-Nh3YXmUBoiwBOJQ_LHm:b-wmhpk3)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @herkolategan and @RaduBerinde)

**Comment by jbowens** [comment]
bors r=DarrylWong

**Pull request merged**